determined-ai/determined
# feat: rolling upgrades v0

## Description

## Test Plan
Configuration:
- Agent reconnect attempts and backoff interval should bumped up to accommodate for longer master downtimes, e.g.: 
```
agent_reconnect_attempts: 25
agent_reconnect_backoff: 5
```
- Set `agent_reattach_enabled: true` on compute resource pools.
- `agent_reconnect_wait` in master must be higher than `agent_reconnect_backoff` to give the agents time after master startup to reconnect. Default is 25 seconds, which should be okay.

Update/upgrade procedure:
1. Launch an experiment
2. Restart master (with new config or new version)
3. Experiment should continue running
4. Restart agent to upgrade, if necessary.

Limitations:
- Only trials are currently can be recovered with zero downtime. No support for notebooks and other task types yet.
- Agent RM only.

## Commentary (optional)

## Checklist

See [Release Note](../docs/release-notes/README.md) for details.
